bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -935,15 +947,23 @@ private ContentPolicy getContentPolicy(@NotNull Resource imageResource) {
     private List<Integer> getAllowedRenditionWidths(@NotNull Resource imageResource) {
         List<Integer> list = new ArrayList<>();
         ContentPolicy contentPolicy = getContentPolicy(imageResource);
+        Designer designer = getDesginer(imageResource);
```

Comment:
Since `adaptTo` is potentially expensive, I'd rather we call `getDesigner` in the `else` branch (when `contentPolicy` is null)